src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/BackupOperation/BackupOperation.cs

```diff
@@ -148,52 +151,62 @@ public BackupConfigInfo CreateBackupConfigInfo(string databaseName)
             return configInfo;
         }
 
+        public string ScriptContent
+        {
+            get
+            {
+                return this.scriptContent;
+            }
+            set
+            {
+                this.scriptContent = value;
+            }
+        }
+
         /// <summary>
         /// Execute backup
         /// </summary>
-        public void PerformBackup()
+        public void Execute(TaskExecutionMode mode)
         {
+            StringBuilder sb = new StringBuilder();
+            SqlExecutionModes oldExecutionMode = this.dataContainer.Server.ConnectionContext.SqlExecutionModes;
+            this.dataContainer.Server.ConnectionContext.SqlExecutionModes = (mode == TaskExecutionMode.Script) ? SqlExecutionModes.CaptureSql: SqlExecutionModes.ExecuteAndCaptureSql;
+            this.dataContainer.Server.ConnectionContext.CapturedSql.Clear();
             this.backup = new Backup();
             this.backup.Database = this.backupInfo.DatabaseName;
             this.backup.Action = this.backupActionType;
             this.backup.Incremental = this.isBackupIncremental;
             this.SetBackupProps();
 
-            if (this.backup.Action == BackupActionType.Files)
+            try
             {
-                IDictionaryEnumerator filegroupEnumerator = this.backupInfo.SelectedFileGroup.GetEnumerator();
-                filegroupEnumerator.Reset();
-                while (filegroupEnumerator.MoveNext())
+                if (this.backup.Action == BackupActionType.Files)
                 {
```

Comment:
Minor: all these giant if blocks really should be their own methods. I know you probably just ported this code over, but makes it hard to read since the method is 150 lines long